docs/api.rst

```diff
@@ -536,10 +536,59 @@ Transport exceptions
 - ``MessageReceiveError``: The transport encountered any other error while trying to receive a message.
 
 
-ASGI Transport
-++++++++++++++
+Redis Gateway Transport
++++++++++++++++++++++++
+
+The ``transport.redis_gateway`` module provides a transport implementation that uses Redis (in simple or Sentinel mode)
+for sending and receiving messages. This is the recommended transport for use with ``pysoa``, as it provides a
+convenient and performant backend for asynchronous service requests.
+
+Standard and Sentinel modes
+***************************
+
+The Redis Gateway transport has two primary modes of operation: in "standard" mode, the channel layer will connect to a
+specified list of Redis hosts, while in "Sentinel" mode, the channel layer will connect to a list of Sentinel hosts and
+use Sentinel to find its Redis hosts.
+
+Configuration
+*************
+
+The Redis Gateway transport takes the following extra keyword arguments for configuration:
+
+- ``backend_type``: Either "redis.standard" or "redis.sentinel" to specify which Redis backend to use (required)
+
+- ``backend_layer_kwargs``: A dictionary of arguments to pass to the backend layer
+
+  + ``connection_kwargs``: A dictionary of arguments to pass to the underlying Redis client
+
+  + ``hosts``: A list of strings (host names / IP addresses) or tuples (host names / IP addresses and ports) for Redis hosts or sentinels to which to connect (will use "localhost" by default)
+
+  + ``redis_db``: The Redis database number to use (a shortcut for specifying ``connection_kwargs['db']``)
+
+  + ``redis_port``: The connection port to use (a shortcut for providing this for every entry in ``hosts``
+
+  + ``sentinel_refresh_interval``: How often, in seconds, master/slave data should be refreshed from the Sentinel (only for type "redis.sentinel") (refreshes every connection by default)
+
+  + ``sentinel_services``: Which Sentinel services to use (only for type "redis.sentinel") (will be auto-discovered from the Sentinel by default)
+
+- ``message_expiry_in_seconds``: How long a message may remain in the queue before it is considered expired and discarded
+
+- ``queue_capacity``: The maximum number of messages a given Redis queue may hold before the transport should stop pushing messages to it (defaults to 10,000)
```

Comment:
That's a heck of a default! I would have gone with something closer to 100. Thinking of the clients making requests to the servers, I's think we just need some small multiple of the number of workers we expect to run in order to smooth out bursts.

This also applies to the server responding to a client though, doesn't it? The value we set here determines how many unread responses a client can have from a single server?

Reply:
The default applies to both the queues clients use to send requests to servers and the queues servers use to respond to clients. This is comparable to the existing behavior, and serves as an argument for defining non-default settings overrides.